dsa.py

```diff
@@ -439,7 +464,7 @@ def dsa(func: ghost_code.Function) -> Function:
             curr_node_incarnations[prog_var] = incarnation_number
         s.incarnations[current_node] = curr_node_incarnations
 
-    apply_insertions(s)
+    apply_insertions(func, s, new_dsa_vars)
```

Comment:
I don't like passing the entire func to apply insertion because we then set ourselves up to shoot ourselves in the foot.